ports/atmel-samd/common-hal/busio/UART.c

```diff
@@ -152,16 +154,22 @@ void common_hal_busio_uart_construct(busio_uart_obj_t *self,
 
     if (rx && receiver_buffer_size > 0) {
         self->buffer_length = receiver_buffer_size;
-        // Initially allocate the UART's buffer in the long-lived part of the
-        // heap.  UARTs are generally long-lived objects, but the "make long-
-        // lived" machinery is incapable of moving internal pointers like
-        // self->buffer, so do it manually.  (However, as long as internal
-        // pointers like this are NOT moved, allocating the buffer
-        // in the long-lived pool is not strictly necessary)
-        self->buffer = (uint8_t *)gc_alloc(self->buffer_length * sizeof(uint8_t), false, true);
-        if (self->buffer == NULL) {
-            common_hal_busio_uart_deinit(self);
-            mp_raise_msg_varg(&mp_type_MemoryError, translate("Failed to allocate RX buffer of %d bytes"), self->buffer_length * sizeof(uint8_t));
+        if (NULL != receiver_buffer) {
+            self->buffer = receiver_buffer;
+        }
+        else {
```

Comment:
(nitpicks)

```suggestion
        if (receiver_buffer != NULL) {
            self->buffer = receiver_buffer;
        } else {
```

Reply:
> Thanks for fixing this up! Two minor comments. Good otherwise.
Thanks for quick response!

Actually, I have done this intentionally.
I know the rest of the code is written in different manner, but still I believe it is valid point.
Should it be described somewhere here?
https://github.com/micropython/micropython/blob/master/CODECONVENTIONS.md

https://en.wikipedia.org/wiki/Yoda_conditions

```c
common-hal/busio/UART.c: In function 'common_hal_busio_uart_construct':
common-hal/busio/UART.c:157:18: error: lvalue required as left operand of assignment
  157 |         if (NULL = receiver_buffer) {
```

```c
common-hal/busio/UART.c: In function 'common_hal_busio_uart_construct':
common-hal/busio/UART.c:157:13: error: suggest parentheses around assignment used as truth value [-Werror=parentheses]
  157 |         if (receiver_buffer = NULL) {
      |             ^~~~~~~~~~~~~~~
```

So if you add parentheses like this:
```c
  157 |         if ((receiver_buffer = NULL)) {
```
then compiler consider this code as valid and such kind of errors are going to be discovered first during debugging.

Reply:
Ok, fine with me.